sap_business_document_processing/document_information_extraction_client/dox_api_client.py

```diff
@@ -0,0 +1,608 @@
+        """
+        Gets the text for all pages of a document
+        :param document_id: The ID of the document
+        :return: A dictionary mapping the page numbers to the text
+        """
+        response = self.get(DOCUMENT_PAGES_TEXT_ENDPOINT.format(document_id=document_id),
+                            log_msg_before=f'Getting text for all pages of document with ID {document_id}',
+                            log_msg_after=f'Successfully got text fot all pages of document with ID {document_id}')
+        return response.json()[API_FIELD_RESULTS]
+
+    def get_request_for_document(self, document_id):
+        """
+        Gets the request of a processed document
+        :param document_id: The ID of the document
+        :return: The request of the document as dictionary
+        """
+        response = self.get(DOCUMENT_ID_REQUEST_ENDPOINT.format(document_id=document_id),
+                            log_msg_before=f'Getting request information for document with ID {document_id}',
+                            log_msg_after=f'Successfully got request information for document with ID {document_id}')
+        return response.json()
+
+    def get_page_dimensions_for_document(self, document_id, page_no: int):
+        """
+        Gets the dimensions of a document page
+        :param document_id: The ID of the document
+        :param page_no: The page number for which to get the dimensions
+        :return: The height and width of the document page as dictionary
+        """
+        response = self.get(DOCUMENT_PAGE_DIMENSIONS_ENDPOINT.format(document_id=document_id, page_number=page_no),
+                            log_msg_before=f'Getting dimensions for page {page_no} of document with ID {document_id}',
+                            log_msg_after=f'Successfully got dimensions for page {page_no} of document with ID {document_id}')
+        return response.json()
+
+    def get_all_dimensions_for_document(self, document_id):
+        """
+        Gets the dimensions of all document pages
+        :param document_id: The ID of the document
+        :return: The height and width of all document pages as dictionary with page number as key
+        """
+        response = self.get(DOCUMENT_PAGES_DIMENSIONS_ENDPOINT.format(document_id=document_id),
+                            log_msg_before=f'Getting dimensions for all pages of document with ID {document_id}',
+                            log_msg_after=f'Successfully got dimensions for all pages od document with ID {document_id}')
+        return response.json()[API_FIELD_RESULTS]
+
+    def post_ground_truth_for_document(self, document_id, ground_truth):
+        """
+        Saves the ground truth for a document
+        :param document_id: The ID of the document
+        :param ground_truth: Path to the ground truth JSON file or an object representing the ground truth
+        :return: The API endpoint response as dictionary
+        """
+        if isinstance(ground_truth, str):
```

Comment:
This can be taken to utility function and reused between DC and DOX

Reply:
done